pandas/io/excel/_openpyxl.py

```diff
@@ -451,3 +462,280 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
                             xcell = wks.cell(column=col, row=row)
                             for k, v in style_kwargs.items():
                                 setattr(xcell, k, v)
+
+
+class _OpenpyxlReader(_BaseExcelReader):
+
+    def __init__(self, filepath_or_buffer):
+        """Reader using openpyxl engine.
+
+        Parameters
+        ----------
+        filepath_or_buffer : string, path object or Workbook
+            Object to be parsed.
+        """
+        err_msg = "Install openpyxl >= 2.4.0 for Excel with Openpyxl support"
+
+        try:
+            import openpyxl
+        except ImportError:
+            raise ImportError(err_msg)
+        else:
+            if openpyxl.__version__ < LooseVersion('2.4.0'):
+                raise ImportError(err_msg +
+                                  ". Current version " + openpyxl.__version__)
+        super().__init__(filepath_or_buffer)
+
+    @property
+    def _workbook_class(self):
+        from openpyxl import Workbook
+        return Workbook
+
+    def load_workbook(self, filepath_or_buffer):
+        from openpyxl import load_workbook
+        return load_workbook(filepath_or_buffer, data_only=True)
+
+    @property
+    def sheet_names(self):
+        return self.book.sheetnames
+
+    @staticmethod
+    def _handle_usecols(frame, usecols):
```

Comment:
What did you add this for? Doesn't seem like functionality specific to openpyxl

Reply:
the xlrd basically hands off all the heavy lifting to `pandas.io.parsers.TextParser`, which makes sense as `xlrd` returns strings. So that is probably why there are many keywords related to specifying exactly how and if the text fields need to be translated to other data types. `Openpyxl` has a better understanding of the excel types and more or less returns the python equivalent for each cell. IMHO doesn't make sense to cast the output back to strings so it can be reinterpreted by a TextParser, so instead I chose to replicate its behaviour. Earlier I proposed to just drop support for some of the keywords as I feel they don't really make sense, but after discussions we decided to prioritize compatibility over a leaner implementation. There are a few edge cases where I couldn't get 100% equivalent output, but I think I got pretty close to 99%

Reply:
OK fair enough. Let's put this on the back burner for now